tools/bus_space/bus.c

```diff
@@ -156,9 +156,12 @@ bs_read(int rid, off_t ofs, void *buf, ssize_t bufsz)
 		case 4:
 			*((uint32_t *)buf) = *((volatile uint32_t *)ptr);
 			break;
+		case 8:
+			*((uint64_t *)buf)[] = *((volatile uint64_t *)ptr)[];
+			break;
 		default:
-			errno = EIO;
 			return (0);
+			errno = EIO;
```

Comment:
errno needs to be set before the return.

Reply:
swapped lines have been pushed.